source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -210,64 +218,138 @@ class UdpProxyFilter : public Network::UdpListenerReadFilter,
     // using it in the context of absl.
     using is_transparent = void; // NOLINT(readability-identifier-naming)
 
+    HeterogeneousActiveSessionHash(const bool consider_host) : consider_host_(consider_host) {}
+
     size_t operator()(const Network::UdpRecvData::LocalPeerAddresses& value) const {
       return absl::Hash<const Network::UdpRecvData::LocalPeerAddresses>()(value);
     }
-    size_t operator()(const ActiveSessionPtr& value) const {
-      return absl::Hash<const Network::UdpRecvData::LocalPeerAddresses>()(value->addresses());
+    size_t operator()(const LocalPeerHostAddresses& value) const {
+      auto hash = this->operator()(value.local_peer_addresses_);
+      if (consider_host_) {
+        hash = absl::HashOf(hash, value.host_.address()->asStringView());
+      }
+      return hash;
     }
     size_t operator()(const ActiveSession* value) const {
-      return absl::Hash<const Network::UdpRecvData::LocalPeerAddresses>()(value->addresses());
+      LocalPeerHostAddresses key{value->addresses(), value->host()};
+      return this->operator()(key);
     }
+    size_t operator()(const ActiveSessionPtr& value) const { return this->operator()(value.get()); }
+
+  private:
+    const bool consider_host_;
```

Comment:
Can't this be even simpler? Just require passing host or nullptr and then just deal with it? Then it would be mostly identical?

Reply:
There is a problem with this approach when we lookup directly using `ActiveSession*` or `ActiveSessionPtr` as this class has always `host` and I didn't want to introduce flag to this class.
If we lookup only using `LocalPeerHostAddresses`, it won't be a problem